### PR TITLE
Tidy AMP packages

### PR DESCRIPTION
**Pull request by nicl**
* remove components/lib
* remove components/primitives

Both the folders are used minimally and I don't think it's clear when they should be used. For example, Elements.tsx returns a React component, so why does it live in `lib` currently?

The latter does reduce some type safety. However, we don't use the sole component (Img) for most of our amp images and it's a lot of code for only a small amount of benefit (it's not too hard given
the fast feedback to work out if you've mispelt an attribute). I'm not convinced at this stage that we will ever type up our amp- components unless a third-party library that is well-supported magically appears. The lack of typing for attributes on these elements has not really proved a problem so far.

Before:

Screenshot 2019-05-22 at 15 58 17

After:
Screenshot 2019-05-22 at 15 58 27

### CODE REVIEW

**Comment by nicl** [comment]
the codeclimate fail relates to existing code (the Elements file which is by necessity a large switch statement) that has just been moved to a different place.

**Comment by shtukas** [comment]
Nice, thanks!

File: packages/frontend/amp/components/elements/InteractiveMarkup.tsx [hunk 1]
```diff
@@ -24,6 +24,25 @@ export const InteractiveMarkup: React.SFC<{
...
+        html.offsetHeight
+    );
+    
+    window.parent.postMessage({
+        sentinel: 'amp',
+        type: 'embed-size',
+        height: height
+    }, '*');
+    
+    console.log(height);
```
Comment (gtrufitt) [inline packages/frontend/amp/components/elements/InteractiveMarkup.tsx]: Did this script sneak in? Looks cool, just checking... (though console.logggy maybe shouldn't be here)

Comment (nicl) [reply packages/frontend/amp/components/elements/InteractiveMarkup.tsx]: Yeah this is a mistake from a previous PR. I'll clean it up!

Comment (nicl) [reply packages/frontend/amp/components/elements/InteractiveMarkup.tsx]: And actually I'm not even sure if this works yet so going to pull it for now.

Comment (shtukas) [reply packages/frontend/amp/components/elements/InteractiveMarkup.tsx]: Oh. Thanks @gtrufitt !

**Pull request merged**